src/Workspaces/Core/Portable/Storage/AbstractPersistentStorageService.cs

```diff
@@ -63,6 +63,9 @@ public ValueTask<IChecksummedPersistentStorage> GetStorageAsync(Workspace worksp
         {
             if (!DatabaseSupported(solutionKey, checkBranchId))
             {
+                if (workspace.Options.GetOption(StorageOptions.DatabaseMustSucceed))
```

Comment:
i added these optiosn for tests and for local experiementation.  because persistence service fails over gracefully it can be hard to tell when something has gone wrong.  With this option enabled, we hard fail, which is great for local testing as you know immediately if there is a problem.

Reply:
I notice in many of the places below it checks mustsucceed before throwing.  In the case that we don't throw, should failures be logged somewhere (nfw, telemetry, something else?)?  If it fails over so gracefully I worry we might otherwise miss legitimate issues

Reply:
That's a good question. I will check to see about that.

Reply:
ok, i checked on things.  we already throw in caes where there is a true invariatn issue.

What's going on here though is that we are resilient to legal things a caller might do.  For example, we may have the persistence service, but the user may just 'new up' a workspace/solution in memory that has no set solution-path.  It's totally legal to call into us, asking us to store data.  In that case we just return 'false' to say "sorry, we couldn't do it".  

THe value of this flag is that i'm effectively telling the system: you're in a context where this sort of behavior *of the caller* should be considered a bug.  It's useful when testing things out so that we can distinguish why we get certain results (for example, did we fail to read a stream because we truly didn't have it, or because the input was wrong).